### PR TITLE
Show instance type metadata in dropdown selection in Head node wizard Page

### PR DESCRIPTION
**Pull request by tmscarla**
## Description

Currently instance type selection in Head node wizard page is implemented leveraging [Autosuggest](https://cloudscape.design/components/autosuggest/?tabId=playground) Cloudscape component.

This PR replaces the `Autosuggest` component with a `Select` component, in order to show instance metadata such as memory, virtual cores and Gpu info upon selection.

## How Has This Been Tested?

* Manually tests on local environment

## References

* https://cloudscape.design/components/select/?tabId=playground

## Screenshots

Screenshot 2023-03-13 at 12 08 21

## PR Quality Checklist

- [ ] I added tests to new or existing code
- [x] I removed hardcoded strings and used [`react-i18next`](https://react.i18next.com/) library ([useTranslation hook](https://react.i18next.com/latest/usetranslation-hook) and/or [Trans component](https://react.i18next.com/latest/trans-component)), see an example [here](https://github.com/aws/aws-parallelcluster-ui/commit/a6f1e2aa46b245b5bf7500a04b83195477a5cfa5)
- [ ] I made sure no sensitive info gets logged at any time in the codebase (see [here](https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html)) (e.g. no user info or details, no stacktraces, etc.)
- [ ] I checked that infrastructure/update_infrastructure.sh runs without any error
- [x] I checked that `npm run build` builds without any error
- [x] I checked that clusters are listed correctly
- [x] I checked that a new cluster can be created (config is produced and dry run passes)
- [ ] I checked that login and logout work as expected

In order to increase the likelihood of your contribution being accepted, please make sure you have read both the [Contributing Guidelines](../CONTRIBUTING.md) and the [Project Guidelines](../PROJECT_GUIDELINES.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: frontend/src/old-pages/Configure/Components.tsx [hunk 1]
```diff
@@ -163,46 +155,60 @@ export function useInstanceGroups(): Record<string, InstanceGroup> {
...
-          setState(path, detail.value)
-          callback && callback(detail.value)
-        }
-      }}
-      // @ts-expect-error TS(2322) FIXME: Type '(newValue: string) => void' is not assignabl... Remove this comment to see the full error message
-      enteredTextLabel={newValue => {
-        if (newValue !== value) {
-          setState(path, newValue)
-          callback && callback(newValue)
+        setSelectedOption(detail.selectedOption as InstanceTypeOption)
```
Comment (mtfranchetto) [inline frontend/src/old-pages/Configure/Components.tsx:201]: todo: replace it with `useCallback`

File: frontend/src/old-pages/Configure/Queues/MultiInstanceComputeResource.tsx [hunk 2]
```diff
@@ -94,10 +95,10 @@ export function ComputeResource({
       Object.keys(instanceGroups).map(groupName => {
         return {
           label: groupName,
-          options: instanceGroups[groupName].map(([value, label]) => ({
-            label: value,
-            description: label,
-            value: value,
+          options: instanceGroups[groupName].map((instance: InstanceType) => ({
```
Comment (mtfranchetto) [inline frontend/src/old-pages/Configure/Queues/MultiInstanceComputeResource.tsx:98]: question: does the `SingleInstanceComputeResource` need to be aligned as well? It is still used if the PC version is 3.2.0

**Review by mtfranchetto — CHANGES_REQUESTED** [review]

File: frontend/src/old-pages/Configure/Components.tsx [hunk 3]
```diff
@@ -163,46 +155,60 @@ export function useInstanceGroups(): Record<string, InstanceGroup> {
...
-  const instanceToOption = ([value, label]: string[]) => {
-    return {label: value, description: label, value: value}
+  const instanceToOption = (instance: InstanceType): InstanceTypeOption => {
+    return {label: instance.type, value: instance.type, tags: instance.tags}
   }
 
+  const instanceTypeToOption = (
+    instanceType: string,
+  ): InstanceTypeOption | null => {
+    debugger
```
[the remark this answers is not in the document]
Comment (tmscarla) [reply frontend/src/old-pages/Configure/Components.tsx:184]: Yes, forgot it while debugging

File: frontend/src/old-pages/Configure/Components.tsx [hunk 4]
```diff
@@ -163,46 +155,60 @@ export function useInstanceGroups(): Record<string, InstanceGroup> {
 
     if (!(group in groups)) groups[group] = []
 
-    let desc = `${instance.VCpuInfo.DefaultVCpus} vcpus, ${
-      instance.MemoryInfo.SizeInMiB / 1024
-    }GB memory`
+    let tags = [
+      `${instance.VCpuInfo.DefaultVCpus} vCore`,
```
[the remark this answers is not in the document]
Comment (tmscarla) [reply frontend/src/old-pages/Configure/Components.tsx:159]: Gotcha

File: frontend/src/old-pages/Configure/Components.tsx (hunk 1 above)
Comment (tmscarla) [reply frontend/src/old-pages/Configure/Components.tsx:201]: Gotcha, will fix in next revision

File: frontend/src/old-pages/Configure/Queues/MultiInstanceComputeResource.tsx (hunk 2 above)
Comment (tmscarla) [reply frontend/src/old-pages/Configure/Queues/MultiInstanceComputeResource.tsx:98]: Checked manually and it works.

It leverages the same `InstanceSelect` component.

File: frontend/src/old-pages/Configure/Components.tsx [hunk 5]
```diff
@@ -135,12 +129,11 @@ function SubnetSelect({value, onChange, disabled}: any) {
...
 }
 
-type InstanceGroup = [string, string][]
-
 export function useInstanceGroups(): Record<string, InstanceGroup> {
+  const {t} = useTranslation()
   const instanceTypes = useState(['aws', 'instanceTypes']) || []
 
-  let groups: {[key: string]: [string, string][]} = {}
+  let groups: {[key: string]: InstanceGroup} = {}
```
Comment (mtfranchetto) [inline frontend/src/old-pages/Configure/Components.tsx:136]: suggestion(non-blocking): to simplify the typing in the future you can use the equivalent `Record<InstanceGroup>`

**Review by mtfranchetto — APPROVED** [review]

**Pull request merged**